packages/engine/src/lib/handler/piece-executor.ts

```diff
@@ -108,6 +108,10 @@ const executeAction: ActionHandler<PieceAction> = async ({ action, executionStat
                 url.search = new URLSearchParams(params.queryParams).toString()
                 return url.toString()
             },
+            generateApprovalUrl: (params) => {
```

Comment:
Adding a method to the engine is irreversible, the resume URL is a generic method